### PR TITLE
Gradle dependencies by Kotlin DSL

### PR DESCRIPTION
**Pull request by benjamin-cheng**
Kotlin + buildsrc = Android Studio Autocomplete & Click Support

Reference: https://handstandsam.com/2018/02/11/kotlin-buildsrc-for-better-gradle-dependency-management/

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.5.1-all.zip
```
Comment (cnevinc) [inline gradle/wrapper/gradle-wrapper.properties]: Is this a must?

Comment (benjamin-cheng) [reply gradle/wrapper/gradle-wrapper.properties]: No. It's ok to have this feature with gradle 4.1. However, the plugin and kotlin version(1.1.3) is pretty old in gradle 4.1. From gradle 4.5, the built-in kotlin version is upgraded to 1.2.0.

**Comment by benjamin-cheng** [comment]
Yes. apply this mechanism doesn't increase the APK size.

**Comment by mTwTm** [comment]
Yeah with the new added script we are confident that this change will not bump APK size (diff is around 1KB)

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (mTwTm) [reply gradle/wrapper/gradle-wrapper.properties]: I think we should update this. Just maybe not before release so maybe merge this after we build beta rc.

**Pull request merged**